apiserver/facades/client/controller/controller.go

```diff
@@ -47,18 +42,43 @@ type ControllerAPIv3 struct {
 	resources  facade.Resources
 }
 
+// ControllerAPIv3 provides the v3 Controller API.
+type ControllerAPIv3 struct {
```

Comment:
The v3 server will now expose all v4 things, which is wrong. Looks like we're doing that in ModelManager, which is doing it wrong.

ControllerAPIv3 already embeds ModelStatusAPI, so you just need to shadow it with the v3 signature - which you're already doing

Reply:
As per our offline discussion, I'll keep this as is. 
I am happy to add interfaces for different versions exposing what calls each version has, similarly to ModelManger. However, I would prefer to do it in a separate PR.